pkg/binary/envoy/debug/pii_test.go

```diff
@@ -0,0 +1,132 @@
+				` Chrome/76.0.3809.132 Safari/537.36'`},
+			outErr: "",
+		},
+		"multiple_pii_fields": {
+			in:     []string{validLog, logWithOneField},
+			format: format,
+			pii: map[string]bool{
+				"[%START_TIME%]": true,
+				"%REQ(:METHOD)% %REQ(X-ENVOY-ORIGINAL-PATH?:PATH)% %PROTOCOL%": true,
+				"%RESPONSE_CODE%": true,
+			},
+			outLogs: []string{`pii pii pii - - 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/76.0.3809.132 Safari/537.36'`},
+			outErr:  "",
+		},
+
+		// negative test cases
+		"empty_logs": {
+			in:      []string{},
+			format:  format,
+			pii:     map[string]bool{"[%START_TIME%]": true},
+			outLogs: []string{},
+			outErr:  "",
+		},
+		"all_invalid_logs": {
+			in:      []string{"test", "test1", "test2"},
+			format:  format,
+			pii:     map[string]bool{"[%START_TIME%]": true},
+			outLogs: []string{},
+			outErr:  "",
+		},
+		"empty_format_string": {
+			in:      []string{validLog, logWithOneField},
+			format:  "",
+			pii:     map[string]bool{"[%START_TIME%]": true},
+			outLogs: []string{},
+			outErr:  "",
+		},
+		"empty_pii_map": {
+			in:     []string{validLog, logWithOneField},
+			format: format,
+			pii:    map[string]bool{},
+			outLogs: []string{`'[2019-09-05T17:53:36.908Z]' 'GET /productpage HTTP/1.1' 200 - - 'Mozilla/5.0 (X11; Linux x86_64)` +
+				` AppleWebKit/537.36 (KHTML, like Gecko) Chrome/76.0.3809.132 Safari/537.36'`},
+			outErr: "",
+		},
+		"invalid_pii_map": {
+			in:     []string{validLog, logWithOneField},
+			format: format,
+			pii:    map[string]bool{"test": true, "test2": true},
+			outLogs: []string{`'[2019-09-05T17:53:36.908Z]' 'GET /productpage HTTP/1.1' 200 - - 'Mozilla/5.0 (X11; Linux x86_64)` +
+				` AppleWebKit/537.36 (KHTML, like Gecko) Chrome/76.0.3809.132 Safari/537.36'`},
+			outErr: "",
+		},
+	}
+
+	piiFunc := func(string) string { return "pii" }
+
+	// defining subtest with t.Run
+	//nolint:gocritic,scopelint
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			actualLogs, actualErr := ProcessLogs(tc.in, tc.format, tc.pii, piiFunc)
+
+			if actualErr != nil {
+				if tc.outErr == "" {
+					// expected no error but get error
+					t.Fatalf("ProcessLogs(%v, %v, %v, piiFunc) = %v, %v, expected no error", tc.in, tc.format, tc.pii, actualLogs, actualErr)
+				} else if !strings.Contains(actualErr.Error(), tc.outErr) {
+					// actual error does not contain the expected error
+					t.Fatalf("ProcessLogs(%v, %v, %v, piiFunc) = %v, %v, but expected error: %v", tc.in, tc.format, tc.pii, actualLogs, actualErr, tc.outErr)
+				}
+			} else if tc.outErr != "" {
+				// expect error but no acutal error
+				t.Fatalf("ProcessLogs(%v, %v, %v, piiFunc) = %v, %v, but expected error: %v", tc.in, tc.format, tc.pii, actualLogs, actualErr, tc.outErr)
+			} else if !reflect.DeepEqual(actualLogs, tc.outLogs) {
```

Comment:
note: one option here is to drop the `else` and just keep this a bare `if !reflect.DeepEqual(actualLogs, ...)` - this means you always check the output even when err != nil, but it can be a decent practice (e.g. to ensure you always return an empty list when you return an error - as written we assert nothing about the returned value when the error is not nil).